src/libs/antares/study/parts/thermal/cluster.cpp

```diff
@@ -421,6 +464,56 @@ void Data::ThermalCluster::calculationOfSpinning()
     }
 }
 
+void Data::ThermalCluster::calculationOfMarketBidPerHourAndMarginalCostPerHour()
+{
+    if (costgeneration == Data::setManually || !prepro)
+        costGenManualCalculationOfMarketBidAndMarginalCostPerHour();
+    else // costgeneration == Data::useCostTimeseries
+        costGenTimeSeriesCalculationOfMarketBidAndMarginalCostPerHour();
```

Comment:
```suggestion
    switch (costgeneration) {
    case Data::setManually:
        costGenManualCalculationOfMarketBidAndMarginalCostPerHour();
        break;
    case Data::useCostTimeseries:
        costGenTimeSeriesCalculationOfMarketBidAndMarginalCostPerHour();
        break;
```
Note that `prepro` is now completely unrelated.